pkg/model/universe.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package model
+
+import (
+	"io/ioutil"
+	"strings"
+
+	"github.com/gobuffalo/flect"
+
+	"sigs.k8s.io/kubebuilder/pkg/scaffold/input"
+	"sigs.k8s.io/kubebuilder/pkg/scaffold/resource"
+	"sigs.k8s.io/kubebuilder/pkg/scaffold/util"
+)
+
+// Universe describes the entire state of file generation
+type Universe struct {
+	Project *input.ProjectFile
+
+	Boilerplate string `json:"boilerplate,omitempty"`
+
+	Resource *Resource `json:"resource,omitempty"`
+
+	Files []*File `json:"files,omitempty"`
+
+	// Multigroup tracks if the project has more than one group
+	MultiGroup bool `json:"multigroup,omitempty"`
+}
+
+// NewUniverse creates a new Universe
+func NewUniverse(options ...UniverseOption) (*Universe, error) {
+	universe := &Universe{}
+
+	// Apply options
+	for _, option := range options {
+		if err := option(universe); err != nil {
+			return nil, err
+		}
+	}
+
+	return universe, nil
+}
+
+// UniverseOption configure Universe
+type UniverseOption func(*Universe) error
+
+// WithProjectFrom loads the project configuration from the provided path
+func WithProjectFrom(path string) UniverseOption {
+	return func(universe *Universe) error {
+		project, err := util.LoadProjectFile(path)
+		if err != nil {
+			return err
+		}
+
+		universe.Project = project
+		return nil
+	}
+}
+
+// WithProject stores the already loaded project configuration
+func WithProject(project *input.ProjectFile) UniverseOption {
+	return func(universe *Universe) error {
+		universe.Project = project
+		return nil
+	}
+}
+
+// WithBoilerplateFrom loads the boilerplate from the provided path
+func WithBoilerplateFrom(path string) UniverseOption {
+	return func(universe *Universe) error {
+		boilerplate, err := ioutil.ReadFile(path)
+		if err != nil {
+			return err
+		}
+
+		universe.Boilerplate = string(boilerplate)
+		return nil
+	}
+}
+
+// WithBoilerplate stores the already loaded project configuration
+func WithBoilerplate(boilerplate string) UniverseOption {
+	return func(universe *Universe) error {
+		universe.Boilerplate = string(boilerplate)
+		return nil
+	}
+}
+
+// WithoutBoilerplate is used for files that do not require a boilerplate
+func WithoutBoilerplate(universe *Universe) error {
+	universe.Boilerplate = "-"
+	return nil
+}
```

Comment:
could not be WithBoilerplate or WithoutBoilerplate automatically/by default?
I mean, could not be `universe.Boilerplate = "-"` by default?  Which means that the above func would not be required at all.